subsys/mgmt/ec_host_cmd/ec_host_cmd_handler.c

```diff
@@ -27,36 +27,30 @@ BUILD_ASSERT(NUMBER_OF_CHOSEN_BACKENDS < 2, "Number of chosen backends > 1");
-		(static uint8_t _name##_rx_buffer[CONFIG_EC_HOST_CMD_HANDLER_RX_BUFFER_SIZE];),    \
-		())                                                                                \
-	COND_CODE_1(                                                                               \
-		CONFIG_EC_HOST_CMD_HANDLER_TX_BUFFER_DEF,                                          \
-		(static uint8_t _name##_tx_buffer[CONFIG_EC_HOST_CMD_HANDLER_TX_BUFFER_SIZE];),    \
-		())                                                                                \
-	static K_KERNEL_STACK_DEFINE(_name##stack, CONFIG_EC_HOST_CMD_HANDLER_STACK_SIZE);         \
-	static struct k_thread _name##thread;                                                      \
-	static struct ec_host_cmd _name = {                                                        \
-		.rx_ctx =                                                                          \
-			{                                                                          \
-				.buf = COND_CODE_1(CONFIG_EC_HOST_CMD_HANDLER_RX_BUFFER_DEF,       \
-						   (_name##_rx_buffer), (NULL)),                   \
-			},                                                                         \
-		.tx =                                                                              \
-			{                                                                          \
-				.buf = COND_CODE_1(CONFIG_EC_HOST_CMD_HANDLER_TX_BUFFER_DEF,       \
-						   (_name##_tx_buffer), (NULL)),                   \
-				.len_max = COND_CODE_1(                                            \
-					CONFIG_EC_HOST_CMD_HANDLER_TX_BUFFER_DEF,                  \
-					(CONFIG_EC_HOST_CMD_HANDLER_TX_BUFFER_SIZE), (0)),         \
-			},                                                                         \
-		.thread = &_name##thread,                                                          \
-		.stack = _name##stack,                                                             \
-	}
-
-EC_HOST_CMD_DEFINE(ec_host_cmd);
+COND_CODE_1(CONFIG_EC_HOST_CMD_HANDLER_RX_BUFFER_DEF,
+	    (static uint8_t hc_rx_buffer[CONFIG_EC_HOST_CMD_HANDLER_RX_BUFFER_SIZE];), ())
+COND_CODE_1(CONFIG_EC_HOST_CMD_HANDLER_TX_BUFFER_DEF,
+	    (static uint8_t hc_tx_buffer[CONFIG_EC_HOST_CMD_HANDLER_TX_BUFFER_SIZE];), ())
+
+#ifdef CONFIG_EC_HOST_CMD_DEDICATED_THREAD
+static K_KERNEL_STACK_DEFINE(hc_stack, CONFIG_EC_HOST_CMD_HANDLER_STACK_SIZE);
+#endif /* CONFIG_EC_HOST_CMD_DEDICATED_THREAD */
+
+static struct ec_host_cmd ec_host_cmd = {
+	.rx_ctx = {
+			.buf = COND_CODE_1(CONFIG_EC_HOST_CMD_HANDLER_RX_BUFFER_DEF, (hc_rx_buffer),
+					   (NULL)),
+		},
+	.tx = {
+			.buf = COND_CODE_1(CONFIG_EC_HOST_CMD_HANDLER_TX_BUFFER_DEF, (hc_tx_buffer),
+					   (NULL)),
+			.len_max = COND_CODE_1(CONFIG_EC_HOST_CMD_HANDLER_TX_BUFFER_DEF,
+					       (CONFIG_EC_HOST_CMD_HANDLER_TX_BUFFER_SIZE), (0)),
+		},
+#ifdef CONFIG_EC_HOST_CMD_DEDICATED_THREAD
+	.stack = hc_stack,
```

Comment:
This is also not needed in the struct right? Just use `hc_stack` directly in `k_thread_create`, saves another pointer.

Reply:
Done.